### PR TITLE
Fix validation error on Batch import

### PR DESCRIPTION
**Pull request by monishdeb**
Fixes https://github.com/JMAConsulting/biz.jmaconsulting.lineitemedit/issues/25

### CODE REVIEW

File: CRM/Lineitemedit/Util.php [hunk 1]
```diff
@@ -988,14 +988,31 @@ public static function generatePriceField() {
...
+          ]
+        ));
+      }
+      else {
+        civicrm_api3('PriceField', 'create', ['id' => key($priceField), 'is_active' => TRUE]);
+      }
+    }
+  }
+
+  public static function disableEnablePriceField($enable = FALSE) {
```
Comment (JoeMurray) [inline CRM/Lineitemedit/Util.php]: Could you improve the name here? Do you mean disablePriceField() ? Or better: disableLineItemEditPriceFields() ?

File: info.xml [hunk 2]
```diff
@@ -15,11 +15,12 @@
     <url desc="Licensing">http://www.gnu.org/licenses/agpl-3.0.html</url>
   </urls>
   <releaseDate>2018-03-06</releaseDate>
-  <version>2.0</version>
+  <version>2.1</version>
   <develStage>stable</develStage>
   <compatibility>
     <ver>4.7</ver>
     <ver>5.0</ver>
+    <ver>5.8</ver>
```
Comment (JoeMurray) [inline info.xml]: Why do we need this? Can we indicate something like 5.0+ ? Also, I think that Eileen has removed 4.7 from new releases of her extensions. People will still be able to get versions that are compatible with 4.7, but she doesn't want to be forced to support them. We might want to do the same...do you want to think about this?

File: lineitemedit.php [hunk 3]
```diff
@@ -161,6 +161,15 @@ function lineitemedit_civicrm_postProcess($formName, &$form) {
...
     }
   }
+  if ('CRM_Batch_Form_Entry' == $formName) {
+    CRM_Lineitemedit_Util::disableEnablePriceField(TRUE);
+  }
+}
+
+function lineitemedit_civicrm_validateForm($formName, &$fields, &$files, &$form, &$errors) {
+  if ('CRM_Batch_Form_Entry' == $formName && empty($errors)) {
+    CRM_Lineitemedit_Util::disableEnablePriceField();
```
Comment (JoeMurray) [inline lineitemedit.php:171]: What if they never successfully validate the form? The PriceFields will be left disabled.

What if others are using the system and need the price fields while someone is using the Batch Form Entry form? I don't think this is a feasible approach. Let's put more effort into dealing with the cause of the issue in that old JIRA issue.

Comment (monishdeb) [reply lineitemedit.php:171]: @JoeMurray actually I have used ```empty($errors)``` which means the PFs got disabled ONLY upon successful submission, in other words, there are no validation error messages. And then re-enabling those PFs after submission that occurs in postprocess hook. There is no halt between these two function call and a very short duration between these calls. 
Alternatively, the only fix I found is to remove the core function in Batch Entry. Will take a look at it, to see if we can safely remove that core fn from Civi.

Comment (JoeMurray) [reply lineitemedit.php:171]: 'Errors set by the underlying form system, e.g. for required fields, can not be unset here.' https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_validateForm/ Can you confirm that even though these types of errors cannot be unset, they are present in the $errors array?

What function do mean when you write ' the core function in Batch Entry'?

Comment (monishdeb) [reply lineitemedit.php:171]: @JoeMurray I have checked again with two required fields in the Batch entry form, set by form level (not Civi) and once the required field got values, upon submission there is no error message set in $errors array and thus $errors is empty. It is only when the code here will get triggered i.e. on successul submission. 

About the second question, the core function is [CRM_Price_BAO_PriceSet::getOnlyPriceFieldID](https://github.com/civicrm/civicrm-core/blob/master/CRM/Batch/Form/Entry.php#L463) which expects the quickconfig default price set of Contribution to have only one price-field but in our extension we are having multiple price fields for this particular price-set to support multiple quick-config line-items. So my fix here is to disable those additional price-fields on validateForm hook to bypass this exception error and reenable them on postprocess to restore them again. This is kinda hackish fix but didn't find any other alternative fix to tackle this issue unless I need to modify that core function and submit a core PR.

File: info.xml (hunk 2 above)
Comment (monishdeb) [reply info.xml]: My bad we now change the existing version to the latest i.e. 5.0 should be changed to 5.11 not 5.0+. Heres an example https://github.com/eileenmcnaughton/nz.co.fuzion.extendedreport/blob/master/info.xml#L20

Comment (JoeMurray) [reply info.xml]: Would the extension work on every release in 4.7.x and 5.x that we know of? If so, in order to encourage adoption, let's on this extension mark current release as supporting all versions until we have something that doesn't work except with newer releases.

Comment (monishdeb) [reply info.xml]: @JoeMurray yes, it does. I have just quickly confirmed on my local so it would be ok to mark extension to the current release.

**Comment by monishdeb** [comment]
Merging this PR now.

**Pull request merged**